compiler/Bond/Parser.hs

```diff
@@ -204,14 +208,17 @@ struct = do
     fields = unique $ braces $ manySortedBy (comparing fieldOrdinal) (field <* semi)
     with params e = e { currentParams = params }
     unique p = do
-        fields <- p
-        case findDuplicates fields of
-            [] -> return fields
+        fields' <- p
+        case findDuplicates fields' of
+            [] -> return fields'
             Field {..}:_ -> fail $ "Duplicate definition of the field with ordinal " ++ show fieldOrdinal
       where
         findDuplicates xs = deleteFirstsBy ordinal xs (nubBy ordinal xs)
         ordinal = (==) `on` fieldOrdinal
 
+manySortedBy :: (a -> a -> Ordering)
+             -> ParsecT s u m a
+            -> ParsecT s u m [a]
```

Comment:
manySortedBy :: (a -> a -> Ordering) -> Parser a -> Parser [a] 

the function is more generic but we don't export it and only use with Bond parser.